src/YieldDisburser.sol

```diff
@@ -106,32 +109,30 @@ contract YieldDisburser is OwnableUpgradeable {
         uint256 _blockTime,
         uint256 _minVotingAmount,
         uint256 _minHoldingDuration,
-        uint256 _maxVotes,
         uint256 _maxPoints,
-        uint48 _minTimeBetweenClaims,
+        uint256 _cycleLength,
         uint48 _lastClaimedTimestamp,
         uint256 _lastClaimedBlockNumber,
         uint256 _precision
     ) public initializer {
-        __Ownable_init(msg.sender);
-
         BREAD = Bread(breadAddress);
-        PRECISION = _precision;
-
-        projects = new address[](_projects.length);
-        for (uint256 i; i < _projects.length; ++i) {
+        uint256 projectLength = _projects.length;
+        projects = new address[](projectLength);
+        for (uint256 i; i < projectLength; ++i) {
             projects[i] = _projects[i];
         }
-
+        projectDistributions = new uint256[](projectLength);
         blockTime = _blockTime;
+        PRECISION = _precision;
         minVotingAmount = _minVotingAmount;
-        minHoldingDuration = _minHoldingDuration * 1 days;
-        minRequiredVotingPower = ((minVotingAmount * minHoldingDuration) * PRECISION) / blockTime;
-        maxVotes = _maxVotes;
+        minHoldingDuration = _minHoldingDuration * 1 days; // must hold for atleast _minVotingHoldingDuration  days
```

Comment:
Please double check all your comments. They are consistently inconsistent with expected style.

`// Start with a capital letter with no superfluous spacing`

This one should be:

`// Must hold for at least \`_minVotingHoldingDuration\` days`

Reply:
tbh this comment and the comment below are redundant because there is already a comment on the variable definition at the top of the contract for both "minHoldingDuration" and "minRequiredVotingPower"

Reply:
addressed by 2e9c094 and others